software/riscv-tools/Makefile

```diff
@@ -41,7 +41,22 @@ SPIKE_REPO        := riscv-isa-sim
 SPIKE_URL         := https://github.com/riscv/riscv-isa-sim.git
 SPIKE_PATCH       := spike.patch
 SPIKE_GCC_PATCH   := spike-gcc.patch
-SPIKE_TAG         := v1.0.0
+SPIKE_TAG         := v1.1.0
+
+# How many thread you want to use compiling tools
+
+ifeq ($(OS),Linux)
+        NUMPROC := $(shell grep -c ^processor /proc/cpuinfo)
+else ifeq ($(OS),Darwin)
+        NUMPROC := $(shell sysctl hw.ncpu | awk '{print $$2}')
+endif
+
+# Only take half as many processors as available
+COMPILING_THREADS := $(shell echo "$(NUMPROC)/2"|bc)
+
+ifeq ($(COMPILING_THREADS),0)
+        COMPILING_THREADS = 1
```

Comment:
I would do 

COMPILING_THREADS ?= $(shell echo "($(NUMPROC)+1)/2"|bc)


which will 1. allow users to override from command line and 2 handle the 1 case more elegantly